src/foam/dao/HTTPSink.java

```diff
@@ -53,6 +57,11 @@ public void put(Object obj, Detachable sub) {
       if ( ! SafetyUtil.isEmpty(bearerToken_) ) {
         conn.setRequestProperty("Authorization", "Bearer " + bearerToken_);
       }
+      if ( ! SafetyUtil.isEmpty(payloadSignature_) ) {
+        Mac sha256_HMAC = Mac.getInstance("HmacSHA256");
+        String hashedSignature = Base64.encodeBase64String(sha256_HMAC.doFinal(payloadSignature_.getBytes("UTF-8")));
```

Comment:
we'll need to retrieve the shared secret from our configuration and use it to generate the hash
you will also need to stringfy the obj to get the data to actually be hashed